src/include/platform/internal/GenericPlatformManagerImpl_POSIX.cpp

```diff
@@ -82,21 +82,45 @@ void GenericPlatformManagerImpl_POSIX<ImplClass>::_LockChipStack()
 {
     int err = pthread_mutex_lock(&mChipStackLock);
     assert(err == 0);
+
+#if defined(CHIP_STACK_LOCK_TRACKING_ENABLED)
+    mChipStackIsLocked        = true;
+    mChipStackLockOwnerThread = pthread_self();
+#endif
 }
 
 template <class ImplClass>
 bool GenericPlatformManagerImpl_POSIX<ImplClass>::_TryLockChipStack()
 {
-    return pthread_mutex_trylock(&mChipStackLock) == 0;
+    bool locked = (pthread_mutex_trylock(&mChipStackLock) == 0);
+#if defined(CHIP_STACK_LOCK_TRACKING_ENABLED)
+    if (locked)
+    {
+        mChipStackIsLocked        = true;
+        mChipStackLockOwnerThread = pthread_self();
+    }
+#endif
+    return locked;
 }
 
 template <class ImplClass>
 void GenericPlatformManagerImpl_POSIX<ImplClass>::_UnlockChipStack()
 {
     int err = pthread_mutex_unlock(&mChipStackLock);
     assert(err == 0);
+#if defined(CHIP_STACK_LOCK_TRACKING_ENABLED)
+    mChipStackIsLocked = false;
```

Comment:
This write needs to happen before the call to pthread_mutex_unlock.

Reply:
oops - fixed.